platforms/core-runtime/build-option/src/main/java/org/gradle/internal/buildoption/Origin.java

```diff
@@ -47,6 +51,18 @@ String hintMessage(String hint) {
         return String.format(" (%s)", hint);
     }
 
+    private static class BuildPropertyOrigin extends Origin {
+        public BuildPropertyOrigin(String value) {
+            super(value);
+        }
+
+        @Override
+        public void handleInvalidValue(String value, String hint) {
+            String message = String.format("Value '%s' given for %s Build property is invalid%s", value, source, hintMessage(hint));
```

Comment:
```suggestion
            String message = String.format("Value '%s' given for %s build property is invalid%s", value, source, hintMessage(hint));
```

Reply:
For the existing [Origin.GradlePropertyOrigin](https://github.com/gradle/gradle/blob/f8ee6cae5bd056483bafec72043744540f532e09/platforms/core-runtime/build-option/src/main/java/org/gradle/internal/buildoption/Origin.java#L57) the `gradle.properties` file was refereed as `Gradle property`, for this reason was considered to keep the aligment with current implementation.